flink-connectors/flink-connector-kafka/pom.xml

```diff
@@ -111,6 +111,15 @@ under the License.
 			<scope>test</scope>
 		</dependency>
 
+		<!-- Required to execute the kafka server for testing. Please change the zookeeper version accordingly when changing the Kafka version
+             https://github.com/apache/kafka/blob/839b886f9b732b151e1faeace7303c80641c08c4/gradle/dependencies.gradle#L122 -->
+		<dependency>
+			<groupId>org.apache.zookeeper</groupId>
+			<artifactId>zookeeper</artifactId>
+			<version>3.5.9</version>
+			<scope>test</scope>
+		</dependency>
+
```

Comment:
I believe @dmvk was looking into dropping Zookeeper 3.4 for 1.15, so would that also solve this issue then?

Reply:
If Kafka needs a specific Zookeeper version then it is better to specify that in the pom; otherwise this can easily break if we change the Zookeeper version.

Whether we drop ZK 3.4 or not is largely independent from this in any case, because that primarily applies to flink-shaded-zookeeper.

It would actually be interesting to know which modules actually rely on vanilla zookeeper; maybe the root pom entry isn't that useful anymore.

Reply:
Honestly, I am not sure whether other versions suffice here. I just know that the `kafka-server` no longer has this dependency therefore we have to include it. I guess other zookeeper versions will also work but this is the one currently included by Kafka therefore I left the link to check the corresponding version.

Reply:
@MartijnVisser for more context on the zookeeper bump - https://issues.apache.org/jira/browse/FLINK-24707